scrape.py

```diff
@@ -261,26 +260,28 @@ def get_heartbeat(api_endpoint, access_token, response, lat, lng, lat_i, lng_i):
     return heartbeat
 
 
-def get_token(username, password):
-    return login_ptc(username, password)
-
-
 def login(lat_i, lng_i):
-    access_token = get_token(global_username, global_password)
-    if access_token is None:
-        raise Exception('[-] Wrong username/password')
+    while True:
+        access_token = login_ptc(global_username, global_password)
+        if access_token is None:
+            print('[-] Login failed, retrying')
+            time.sleep(1)
+            continue
```

Comment:
I think maybe `login` should return a session object, that'd remove the global and be closer to how I'd expect a session to work.

Also I think you've changed the semantics here, what does incorrect password do now?